### PR TITLE
Add dockerfiler & docker compose

### PR DESCRIPTION
**Pull request by Jonathan-Adly**
Hey Abi, 

I went ahead and put dockerfile and docker-compose for the repo per this issue:[Add a Dockerfile](https://github.com/abi/screenshot-to-code/issues/5). I also ran into a bunch of issues with port 7000 on a mac. So, I put an env.example to show folks how to easily change the port.

Appreciate you taking a look at this!

### CODE REVIEW

**Comment by abi** [comment]
Thanks for this! Will update the port to 7001 as well for non-Docker users.

File: frontend/Dockerfile [hunk 1]
```diff
@@ -0,0 +1,19 @@
+FROM node:20.9-bullseye-slim
+
+# Set the working directory in the container
+WORKDIR /app
+
+# Copy package.json and yarn.lock
+COPY package.json yarn.lock /app/
+
+# Install dependencies
+RUN yarn install
```
[the remark this answers is not in the document]
Comment (Jonathan-Adly) [reply frontend/Dockerfile:10]: Hey Juan, yes. This PR is for running it locally/for development.

For production, there would be a few changes and some of them would be specific to the platform/deployment approach. 

Decisions like if we need an Nginx proxy, is the frontend being built and hosted on S3 separately, etc.

If there is a specific approach (aka deploy to render button or something along these lines), happy to modify.

My goal was to have anyone spin the repo up by a single command on their machine.

File: docker-compose.yml [hunk 2]
```diff
@@ -0,0 +1,27 @@
+version: '3.9'
+
+services:
+  backend:
+    build: 
+      context: ./backend
+      dockerfile: Dockerfile
+    
+    env_file:
+      - .env
```
[the remark this answers is not in the document]
Comment (Jonathan-Adly) [reply docker-compose.yml:10]: Yes - the original structure was to have two env files. One for the frontend (for the backend url - optional) and one at root. (For the OpenAI key - not optional)

I didn't want to change any of the frontend code; so I kept it the same. Only added the docker bits.

The readme directions under the docker section explain this.

To change to one env file - some of the code would to change.

**Comment by abi** [comment]
Thanks for this @Jonathan-Adly updated the default port to be 7001 as well with this PR.

**Comment by dominicdev** [comment]
hello im very new to docker is how I can I at least make image in docker then test

**Pull request merged**